### ISSUE TITLE
gchart trendlines, other than scatterplots

### ISSUE THREAD

**Issue by aodl** [nicolaskruchten/pivottable#916]
Although pivot table makes it nice and easy to user trend lines for gchart scatterplots, unfortunately it inadvertently disables trendlines for line charts and bar charts by converting date or numeric column data (which are continuous) into strings (which are treated as discrete). 

gCharts only listens to the trendlines option passed in if the data type for both axis is explicitly numeric, date, datetime or timeofday. The only reason this conversion to string is performed by pivot table is to get the number of digits using string.length, to know whether or not to display the column labels as slanted or flat.

So a very simple fix is to change line 82 in gchart_renderer.js from
```
            numCharsInHAxis += row[0].length;
```
to 
```
            if (!isNaN(parseFloat(row))) {
              row = [parseFloat(row)];
              numCharsInHAxis += row[0].toFixed().length;
            } else {
              numCharsInHAxis += row[0].length;
            }
```

The limitation of this is that it only works properly if there is one column that has been dragged onto the pivot table, due to column combos being converted into string colKeys separated by '-'.

I think this deserves a better fix that can handle these different scenarios.

**Comment by aodl** [comment]
As an update to this, an even better solution would be to replace the above with;
```
            if (colKey.length == 1) {
              row = [colKey[0]];
              if (!isNaN(Number(row))) {
                row = [Number(row)];
                numCharsInHAxis += row[0].toFixed().length;
              } else if (new Date(row).toString() != "Invalid Date") {
                row = [new Date(row)];
                numCharsInHAxis += row[0].toString().length;
              } else {
                row = [{ v: i + 1, f: colKey.join("-") }];
                numCharsInHAxis += row[0].f.length;
              }
            } else {
              row = [{ v: i + 1, f: colKey.join("-") }];
              numCharsInHAxis += row[0].f.length;
            }
```

I did a bit of digging into the gchart api, and you can specify the value and format separately. The only issue is that the value is what is displayed along the x axis, while the formatted string is only visible if you hover your mouse over the chart lines. In the case above the value is always just the arbitrary column index

**Comment by nicolaskruchten** [comment]
Yes, it's a limitation of the current system that everything is treated as a string. Would be nice to someday get a chance to tackle #191

**Comment by elbeef** [comment]
This is great thank you for the workaround.
